lib/src/utils/configuration.dart

```diff
@@ -11,6 +11,8 @@ abstract class Config {
 
   String get state;
 
+  List<String>? get scope;
```

Comment:
Hey thanks for submitting PR first of all. 

Now, instead of string here it would be nice to have an enum type of the scope so that client cannot pass your or my name :] 

`scope?.join('%20')` this part should be handled inside abstract class since its repeating over the class which means that we will not implement, we will extend config class.

Reply:
@d3xt3r2909 pushed changes except the enum suggestion. Please let me know your feedback.

Reply:
The thing is that if we add scopes in string then whoever is using this API can provide whatever they want, and I really don't want that people are opening issue because they put `r_emailaddresss` instead of `r_emailaddress`. You was probably using old implementation if you had 7 scopes but currently there are only three (please take a look at pic) [Add Scope/Product in LinkedIn section](https://docs.pathfix.com/integrating-with-linkedin#lUj1b).

Now, regarding enum flexibility issue, to be honest I'm expecting more open issues with missed letters then frequent changing of linkedin API scope, but we can resolve this by: 

```dart

abstract class LinkedInScopes {
  String get emailAddress;

  String get liteProfile;

  String get memberSocial;
}

class LinkedInSupportedScopes extends LinkedInScopes {
  @override
  String get emailAddress => 'r_emailaddress';

  @override
  String get liteProfile => 'r_liteprofile';

  @override
  String get memberSocial => 'w_member_social';
}
```

In this way people could use interface to create new scopes in their own responsibility, or if it brakes they could override `LinkedInSupportedScopes` or create new one, until this lib is not updated

Reply:
@d3xt3r2909  take a look [here](https://learn.microsoft.com/en-us/linkedin/shared/authentication/getting-access). There are more than 15 scopes mentioned.
I will update the code as suggested with the above scopes.

Reply:
i think enum will be better

Reply:
@d3xt3r2909  take a look [here](https://learn.microsoft.com/en-us/linkedin/shared/authentication/getting-access). There are more than 15 scopes mentioned. Not just the open scope there are marketing and sales scopes as well. In my use case I am using marketing scope as well.
please suggest

Reply:
The issue with marketing permission is that I cannot get access for this one so cannot provide mapping for response. Because of that this lib is not supporting this feature unfortunately (its not just about supporting scope - [issue](https://github.com/d3xt3r2909/linkedin_login/issues/63)). 

Anyhow with above example code that I posted this would be resolved and anyone could provide unsupported scope types if for some reason they wish.

Reply:
@d3xt3r2909 can you please elaborate on you solution, how it will work from client perspective. What all details will client need to send?

Reply:
hi @d3xt3r2909 can you please help with the above doubt?

Reply:
Hey yes sure, https://gist.github.com/d3xt3r2909/91c6fbd52ca6ff6ee6be194ef05d5763 

Let me know if something is not clear enough since I don't have laptop with me, its mobile code :D

Reply:
And on client side if you want to use it as some custom permission would be 

class MyPermission extends Scopes {
  const MyPermission() : super('MyPermission');
}

Reply:
@d3xt3r2909 updated the PR. Please check.

Reply:
Hi @d3xt3r2909 any update on this?

Reply:
Hi @d3xt3r2909 any update on this?

Reply:
Hey, sry didn't forget it, just didn't have a time to look at it still. But in next few days I will

Reply:
Hi @d3xt3r2909, did you get time to check the PR?

Reply:
Hey, I already wrote some comments, probably final once?